hton.h

```diff
@@ -163,6 +163,14 @@ unpack_int32(const char *buf)
     return (int32_t)apg_ntoh32(nx);
 }
 
+static inline uint32_t
+unpack_uint32(const char *buf)
+{
+    uint32_t nx;
+    memcpy((char *)&nx, buf, sizeof(uint32_t));
+    return (uint32_t)apg_ntoh32(nx);
```

Comment:
Wouldn't this do the same?

```suggestion
    return (uint32_t)unpack_int32(buf);
```

Reply:
If it is, it should be other way around (apt_ntoh32 returns unsigned, so casting it to signed first and then back is weird).

Reply:
If you're advocating for purity, then it would be best if signed helpers are implemented in terms of unsigned to avoid copy-pasting code:

```c
static inline uint32_t
unpack_uint32(const char *buf)
{
    uint32_t nx;
    memcpy((char *)&nx, buf, sizeof(uint32_t));
    return apg_ntoh32(nx);
}

static inline int32_t
unpack_int32(const char *buf)
{
    return (int32_t)unpack_uint32(buf);
}
```

Reply:
Yes. `unpack_uint32` will be inlined. @tailhook please add unsigned variants for in16 & int64 too.

Reply:
No problem. But while we're at it, why we don't use libc's `ntohs` and `ntohl`?